utils/appliance/implementations/ui.py

```diff
@@ -323,27 +320,41 @@ def do_nav(self, _tries=0, *args, **kwargs):
             logger.info("evmserverd restart requested")
             self.appliance.restart_evm_service()
             self.appliance.wait_for_web_ui()
+            self.go(_tries, *args, **go_kwargs)
 
         if recycle or restart_evmserverd:
             self.appliance.browser.quit_browser()
             logger.debug('browser killed on try {}'.format(_tries))
             # If given a "start" nav destination, it won't be valid after quitting the browser
-            self.go(_tries)
+            self.go(_tries, *args, **go_kwargs)
+
+    def pre_navigate(self, *args, **kwargs):
+        pass
+
+    def post_navigate(self, *args, **kwargs):
+        pass
 
     def log_message(self, msg):
         logger.info("[UI-NAV/{}/{}]: {}".format(self.obj.__class__.__name__, self._name, msg))
 
     def go(self, _tries=0, *args, **kwargs):
+        nav_args = {'use_resetter': True}
+
+        if _tries > 2:
+            # Need at least three tries:
+            # 1: login_admin handles an alert or CannotContinueWithNavigation appears.
+            # 2: Everything should work. If not, NavigationError.
+            raise exceptions.NavigationError(self._name)
+
         _tries += 1
-        use_resetter = True
-        if 'use_resetter' in kwargs:
-            use_resetter = kwargs.pop('use_resetter')
-        self.appliance.browser.widgetastic.dismiss_any_alerts()
-        self.pre_navigate(_tries, *args, **kwargs)
+        for arg in nav_args:
```

Comment:
maybe its fine to just use `nav_args = {'use_resetter': kwargs.pop('use_resetter', True)}` until this loops turns necessary